### PR TITLE
Fix an issue with symfony groups

### LINKED ISSUE #1498: Symfony serializer not working on version >= 1.8
Trying version 1.8 break symfony serialization. Objects get serialized to [] .

### PR DESCRIPTION
**Pull request by magnetik**
This time by initializing groups to null, and set it as array only when needed.

I also changed the `JMSSerializerAdapter` so it has the same behavior.

### CODE REVIEW

File: Context/Context.php [hunk 1]
```diff
@@ -127,6 +127,9 @@ public function getVersion()
      */
     public function addGroup($group)
     {
+        if (!is_array($this->groups)) {
```
Comment (Ener-Getick) [inline Context/Context.php]: `null ===`

File: Serializer/JMSSerializerAdapter.php [hunk 2]
```diff
@@ -88,7 +88,7 @@ private function convertContext(Context $context, $direction)
             $jmsContext->setVersion($context->getVersion());
         }
         $groups = $context->getGroups();
-        if (!empty($groups)) {
+        if ($groups !== null) {
```
Comment (Ener-Getick) [inline Serializer/JMSSerializerAdapter.php]: You should use the Yoda notation `null !==`

File: Context/Context.php [hunk 3]
```diff
@@ -127,6 +127,9 @@ public function getVersion()
      */
     public function addGroup($group)
```
Comment (Ener-Getick) [inline Context/Context.php]: the php doc should be updated to `string|null`

File: Serializer/JMSSerializerAdapter.php [hunk 4]
```diff
@@ -88,7 +88,7 @@ private function convertContext(Context $context, $direction)
             $jmsContext->setVersion($context->getVersion());
         }
         $groups = $context->getGroups();
-        if (!empty($groups)) {
+        if (null !== $groups) {
```
Comment (Ener-Getick) [inline Serializer/JMSSerializerAdapter.php]: can be changed `null !== $context->getGroups()`

File: Serializer/SymfonySerializerAdapter.php [hunk 5]
```diff
@@ -59,7 +59,10 @@ private function convertContext(Context $context)
             $newContext[$key] = $value;
         }
 
-        $newContext['groups'] = $context->getGroups();
+        $groups = $context->getGroups();
+        if (null !== $groups) {
```
Comment (Ener-Getick) [inline Serializer/SymfonySerializerAdapter.php]: same here

**Comment by Ener-Getick** [comment]
Could you add a note in a new `UPGRADING-2.1.md` file please?

**Comment by sstok** [comment]
Could everyone please use descriptive subjects like `Fix Symfony serializer not working on version >= 1.8` instead of `Fix #1498`! there are over 9 pull-requests with these type of subjects and I'm getting tired having to open them see what's it about.

Thanks 👍

**Comment by Ener-Getick** [comment]
Thank you @magnetik ! :)

**Pull request merged**